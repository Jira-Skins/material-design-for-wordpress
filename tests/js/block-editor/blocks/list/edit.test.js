/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Edit from '../../../../../assets/src/block-editor/blocks/list/edit';

/**
 * Shallow render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Edit { ...props } /> );
};

const baseProps = {
	attributes: {
		style: 'basic',
		iconPosition: 'leading',
		iconSize: 'small',
		items: [
			{
				primaryText: 'List Item #1',
				secondaryText: 'List Item #1 secondary text',
				icon: 'spa',
				url: 'http://example.com',
				target: '_blank',
			},
			{
				primaryText: 'List Item #2',
				secondaryText: 'List Item #2 secondary text',
				icon: 'book',
				url: 'http://example.com/item-2',
				target: '',
			},
		],
	},
};

describe( 'Edit', () => {
	beforeAll( () => {
		window.getSelection = () => {
			return {
				addRange: () => {},
				removeAllRanges: () => {},
			};
		};
		document.createRange = () => ( {
			collapse: () => {},
			setStart: () => {},
			setEnd: () => {},
			commonAncestorContainer: {
				nodeName: 'BODY',
				ownerDocument: document,
			},
		} );
	} );

	it( 'matches snapshot with basic style', () => {
		const wrapper = setup( { ...baseProps } );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with two-line style', () => {
		const wrapper = setup( {
			...baseProps,
			attributes: { ...baseProps.attributes, style: 'two-line' },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with trailing icon position', () => {
		const wrapper = setup( {
			...baseProps,
			attributes: { ...baseProps.attributes, iconPosition: 'trailing' },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with no icon position', () => {
		const wrapper = setup( {
			...baseProps,
			attributes: { ...baseProps.attributes, iconPosition: 'none' },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with large icon size', () => {
		const wrapper = setup( {
			...baseProps,
			attributes: { ...baseProps.attributes, iconSize: 'large' },
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
