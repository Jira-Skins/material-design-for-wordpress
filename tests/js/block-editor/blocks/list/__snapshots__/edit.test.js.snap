// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit matches snapshot with basic style 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <ul
        class="mdc-list"
      >
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            favorite
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1
              </div>
            </span>
          </span>
        </li>
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            book
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2
              </div>
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mdc-list"
    >
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          favorite
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1
            </div>
          </span>
        </span>
      </li>
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          book
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2
            </div>
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Edit matches snapshot with large icon size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <ul
        class="mdc-list mdc-list--avatar-list"
      >
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            favorite
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1
              </div>
            </span>
          </span>
        </li>
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            book
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2
              </div>
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mdc-list mdc-list--avatar-list"
    >
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          favorite
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1
            </div>
          </span>
        </span>
      </li>
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          book
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2
            </div>
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Edit matches snapshot with no icon position 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <ul
        class="mdc-list"
      >
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1
              </div>
            </span>
          </span>
        </li>
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2
              </div>
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mdc-list"
    >
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1
            </div>
          </span>
        </span>
      </li>
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2
            </div>
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Edit matches snapshot with trailing icon position 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <ul
        class="mdc-list"
      >
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1
              </div>
            </span>
          </span>
          <span
            class="mdc-list-item__meta material-icons"
          >
            favorite
          </span>
        </li>
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2
              </div>
            </span>
          </span>
          <span
            class="mdc-list-item__meta material-icons"
          >
            book
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mdc-list"
    >
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1
            </div>
          </span>
        </span>
        <span
          class="mdc-list-item__meta material-icons"
        >
          favorite
        </span>
      </li>
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2
            </div>
          </span>
        </span>
        <span
          class="mdc-list-item__meta material-icons"
        >
          book
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Edit matches snapshot with two-line style 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <ul
        class="mdc-list mdc-list--two-line"
      >
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            favorite
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1
              </div>
            </span>
            <span
              class="mdc-list-item__secondary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #1 secondary text
              </div>
            </span>
          </span>
        </li>
        <li
          class="mdc-list-item"
        >
          <span
            class="mdc-list-item__graphic material-icons"
          >
            book
          </span>
          <span
            class="mdc-list-item__text"
          >
            <span
              class="mdc-list-item__primary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2
              </div>
            </span>
            <span
              class="mdc-list-item__secondary-text"
            >
              <div
                aria-multiline="true"
                class="rich-text rich-text block-editor-rich-text__editable"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                List Item #2 secondary text
              </div>
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mdc-list mdc-list--two-line"
    >
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          favorite
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1
            </div>
          </span>
          <span
            class="mdc-list-item__secondary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #1 secondary text
            </div>
          </span>
        </span>
      </li>
      <li
        class="mdc-list-item"
      >
        <span
          class="mdc-list-item__graphic material-icons"
        >
          book
        </span>
        <span
          class="mdc-list-item__text"
        >
          <span
            class="mdc-list-item__primary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2
            </div>
          </span>
          <span
            class="mdc-list-item__secondary-text"
          >
            <div
              aria-multiline="true"
              class="rich-text rich-text block-editor-rich-text__editable"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              List Item #2 secondary text
            </div>
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
